src/dependencies.jl

```diff
@@ -603,19 +613,20 @@ function check_system_handle!(ret,dep,handle)
 end
 
 # Default installation method
-if OS_NAME == :Darwin
+if is_apple()
     defaults = [Binaries,PackageManager,SystemPaths,BuildProcess]
-elseif OS_NAME == :Linux
+elseif is_linux()
     defaults = [PackageManager,SystemPaths,BuildProcess]
-elseif OS_NAME == :Windows
+elseif is_windows()
     defaults = [Binaries,PackageManager,SystemPaths]
 else
     defaults = [SystemPaths,BuildProcess]
 end
 
 function applicable(dep::LibraryDependency)
     if haskey(dep.properties,:os)
-        if (dep.properties[:os] != OS_NAME && dep.properties[:os] != :Unix) || (dep.properties[:os] == :Unix && !Base.is_unix(OS_NAME))
+        if (dep.properties[:os] != Compat.KERNEL && dep.properties[:os] != :Unix) ||
```

Comment:
Compat.KERNEL is not the right thing to compare against for BinDeps backwards compatibility